logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -197,4 +199,65 @@
       expect(subject.uptime).to be >= 0
     end
   end
+
+  context "metrics after config reloading" do
+    let(:dummy_output) { DummyOutput.new }
+    let(:config) { "input { generator { } } output { dummyoutput { } }" }
+    let(:new_config_generator_counter) { 50 }
+    let(:new_config) { "input { generator { count => #{new_config_generator_counter} } } output { dummyoutput {} }" }
+    let(:config_path) do
+      f = Stud::Temporary.file
```

Comment:
not related with this PR, just a comment for feature interactions.

We should be able to actually inject this configuration into the pipeline as plain string and not from a temporal path, what do you think? If we do so the pipeline testing and reasoning will be much easier, as then the agent can reinject the new plain string config on it.

what do you think?

//cc @jsvd